smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/ServiceTrait.java

```diff
@@ -246,6 +298,17 @@ public ServiceTrait build(ShapeId target) {
             return new ServiceTrait(this);
         }
 
+        /**
+         * Sets the target shape to which the trait is applied.
+         *
+         * @param target the ShapeId targeted by the trait.
+         * @return Returns the builder.
+         */
+        public Builder target(ShapeId target) {
```

Comment:
this will always be overwritten

Reply:
Not if I build the object without passing a `ShapeId` to the builder, such as `serviceTrait.toBuilder().build()`. I can make it private though (: